backend/tests/v5/e2e/routes/teamspaces/projects/projects.test.js

```diff
@@ -71,6 +71,155 @@ const testGetProjectList = () => {
+			expect(res.body.code).toEqual(templates.projectNotFound.code);
+		});
+
+		test('should fail if the project data are not valid', async () => {
+			const res = await agent.patch(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: 123 }).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the project name is taken by another project', async () => {
+			// create test project
+			const res = await agent.post(`/v5/teamspaces/${teamspace}/projects/?key=${tsAdmin.apiKey}`)
+				.send({ name: 'Existing Name' }).expect(templates.ok.status);
+
+			const projectsRes = await agent.patch(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: 'Existing Name' }).expect(templates.invalidArguments.status);
+			expect(projectsRes.body.code).toEqual(templates.invalidArguments.code);
+
+			// Delete test project afterwards
+			await agent.delete(`/v5/teamspaces/${teamspace}/projects/${res.body._id}?key=${tsAdmin.apiKey}`)
+				.expect(templates.ok.status);
+		});
+
+		test('should edit project if project name is the same', async () => {
+			await agent.patch(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: testProject.name }).expect(templates.ok.status);
+
+			const res = await agent.get(`/v5/teamspaces/${teamspace}/projects?key=${tsAdmin.apiKey}`).expect(templates.ok.status);
+			expect(res.body.projects.find((p) => p.name === testProject.name)).not.toBe(undefined);
+		});
+
+		test('should edit project if project data are valid', async () => {
+			await agent.patch(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: 'New Name' }).expect(templates.ok.status);
+
+			const res = await agent.get(`/v5/teamspaces/${teamspace}/projects?key=${tsAdmin.apiKey}`).expect(templates.ok.status);
+			expect(res.body.projects.find((p) => p.name === testProject.name)).toBe(undefined);
+			expect(res.body.projects.find((p) => p.name === 'New Name')).not.toBe(undefined);
+
+			// edit the project again to keep it the same for the next tests
+			await agent.patch(`${route}?key=${tsAdmin.apiKey}`)
+				.send({ name: testProject.name }).expect(templates.ok.status);
+		});
+	});
+};
+
+const testDeleteProject = () => {
+	describe('Delete project', () => {
+		const route = (projectId) => `/v5/teamspaces/${teamspace}/projects/${projectId}`;
+
+		test('should fail without a valid session', async () => {
+			const res = await agent.delete(route(testProject._id)).expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail without a valid teamspace', async () => {
+			const res = await agent.delete(`/v5/teamspaces/badTSName/projects/${testProject._id}?key=${tsAdmin.apiKey}`).expect(templates.teamspaceNotFound.status);
+			expect(res.body.code).toEqual(templates.teamspaceNotFound.code);
+		});
+
+		test('should fail if the user is not teamspace admin', async () => {
```

Comment:
Can we change this test to a projectAdmin of the project?

unlicensed user doesnt' even have access to the teamspace in general so it's not quite an accurate test

(applies to any of the not teamspace admin tests)